controllers/metal3remediation_controller_test.go

```diff
@@ -0,0 +1,229 @@
+		m.EXPECT().SetLastRemediationTime(gomock.Any())
+		m.EXPECT().IncreaseRetryCount()
+	} else {
+		m.EXPECT().SetRebootAnnotation(context.TODO()).MaxTimes(0)
+		m.EXPECT().SetLastRemediationTime(gomock.Any()).MaxTimes(0)
+		m.EXPECT().IncreaseRetryCount().MaxTimes(0)
+	}
+
+	if tc.ExpectToStartMachineDeletion {
+		// Go to deleting remediation phase, hand over control of the Machine to CAPI
+		// and annotate the node as unhealthy.
+		m.EXPECT().SetRemediationPhase(capm3.PhaseDeleting)
+		m.EXPECT().SetOwnerRemediatedConditionNew(context.TODO())
+		m.EXPECT().SetUnhealthyAnnotation(context.TODO())
+	} else {
+		m.EXPECT().SetRemediationPhase(capm3.PhaseDeleting).MaxTimes(0)
+		m.EXPECT().SetOwnerRemediatedConditionNew(context.TODO()).MaxTimes(0)
+		m.EXPECT().SetUnhealthyAnnotation(context.TODO()).MaxTimes(0)
+	}
+
+	bmh := &v1alpha1.BareMetalHost{}
+	if tc.GetUnhealthyHostFails {
+		m.EXPECT().GetUnhealthyHost(context.TODO()).Return(nil, nil, fmt.Errorf("can't find foo_bmh"))
+		return m
+	} else {
+		m.EXPECT().GetUnhealthyHost(context.TODO()).Return(bmh, nil, nil)
+	}
+
+	// If user has set bmh.Spec.Online to false, do not try to remediate the host and set remediation phase to failed
+	if tc.HostStatusOffline {
+		m.EXPECT().OnlineStatus(bmh).Return(false)
+		m.EXPECT().SetRemediationPhase(capm3.PhaseFailed)
+		return m
+	} else {
+		m.EXPECT().OnlineStatus(bmh).Return(true)
+	}
+
+	m.EXPECT().GetRemediationType().Return(capm3.RebootRemediationStrategy)
+	m.EXPECT().GetRemediationPhase().Return(tc.RemediationPhase).MinTimes(1)
+
+	if tc.RemediationPhase == capm3.PhaseRunning {
+		if tc.IsLastRemediatedTimeNil {
+			// No remediation time has been set on the host, so make sure the reboot annotation is added
+			m.EXPECT().GetLastRemediatedTime().Return(nil)
+		} else {
+			m.EXPECT().GetLastRemediatedTime().Return(&metav1.Time{Time: time.Now()})
+		}
+
+		if tc.IsRetryLimitZero {
+			m.EXPECT().RetryLimitIsSet().Return(false)
+			// When there's no retrying to do, we don't do remediation and instead go to waiting phase
+			m.EXPECT().SetRemediationPhase(capm3.PhaseWaiting)
+			return m
+		} else {
+			m.EXPECT().RetryLimitIsSet().Return(true)
+		}
+		m.EXPECT().HasReachRetryLimit().Return(false)
+		m.EXPECT().GetTimeout().Return(&metav1.Duration{Duration: time.Minute})
+
+		if tc.IsTimeToRemediate {
+			m.EXPECT().TimeToRemediate(time.Minute).Return(true, time.Duration(0))
+		} else {
+			m.EXPECT().TimeToRemediate(time.Minute).Return(false, time.Minute)
+		}
+	}
+	if tc.RemediationPhase == capm3.PhaseWaiting {
+		m.EXPECT().GetTimeout().Return(&metav1.Duration{Duration: time.Minute})
+		if tc.IsTimeToRemediate {
+			m.EXPECT().TimeToRemediate(time.Minute).Return(true, time.Duration(0))
+			// If we're here, we went into PhaseWaiting because the retry limit was hit on the loop
+			// which repeatedly sets the reboot annotation.
+		} else {
+			m.EXPECT().TimeToRemediate(time.Minute).Return(false, time.Minute)
+		}
+	}
```

Comment:
how about something like, i.e:
```
if tc.RemediationPhase == capm3.PhaseWaiting || tc.RemediationPhase == capm3.PhaseRunning {
    
    if tc.IsTimeToRemediate {
	    m.EXPECT().TimeToRemediate(time.Minute).Return(true, time.Duration(0))
    } else {
	    m.EXPECT().TimeToRemediate(time.Minute).Return(false, time.Minute)
    }
}
```
so we could get rid of the lines 118-122?

Reply:
Nice suggestion, I've implemented it.